### DISCUSSION TITLE
More argopy/python export functions at:  https://dataselection.euro-argo.eu

### DISCUSSION THREAD

**Discussion by gmaze** [euroargodev/argopy#197] [Ideas]
The new https://dataselection.euro-argo.eu is awesome and powered by a fast and scalable server and db. 
We should be able to use it with argopy, as we started to explore in #47 

But the "export" functionality are still clunky, since it relies on a batch job launched in the Ifremer HPC Datarmor. Users receive data, many minutes after export, once the job has passed in the Datarmor queue.

I'd like to make the following 2 propositions, please vote or promote if you think this could be useful !

**Comment by gmaze** [comment]
# 💡 New export button in main page
What if users could simply get a small code snippet in the export popup window showing how to download their current data selection with argopy ?
This would let users decide where and how to get the data from.
![Screenshot 2022-03-24 at 15 14 53](https://user-images.githubusercontent.com/1956032/159935812-1e6e581f-65d3-45d7-aac9-d270c52932ef.png)

**Reply by tcarval** [reply]
I like very much this idea.
We'll have to check that Argopy is able to take into account the rich filters of the Argo data selection.
The subsetting batch that generates NetCDF or CSV receives a list of profiles and generate data files from this list.
Argopy may have to be enhanced for this type of data selection.

We may also decide to limit to simple filters (spatial, temporal, parameters)
![Annotation 2022-03-25 181513](https://user-images.githubusercontent.com/19821468/160169510-948eb82b-5041-47c6-8134-a4b084437ab7.png)

**Reply by gmaze** [reply]
Thanks @tcarval for looking at this
About data selection available in **argopy**:
- [x] space/time, including max pressure (although the polygon is still ongoing work at #169 )
- [x] Position/Data QC and Network: this is possible with the data fetchers options ``mode`` and ``ds`` selecting the standard/expert user modes and the dataset to use (called *network* here)
- [ ] parameters & deployment year: nope, we fetch data from what's available. But it would be manageable to add the code snippet to run after data fetching to enforce these criteria

**Reply by gmaze** [reply]
I guess the most problematic scenario is when data selection comes from the "everywhere" search box.
In this case, we could imagine to create a new data fetcher in **argopy** that would work in 2steps:
1. Get the list of profiles ID corresponding to the search criteria (using https://dataselection.euro-argo.eu/api/full-search-response)
2. Load profiles

With such a fetcher, it would be trivial to "translate" the user data selection into a code snippet. 
It could look like this:

```python
from argopy import DataFetcher
argo = DataFetcher(src='ea').selection(search="coatanoan", where="everywhere")
ds = argo.load().data
```

But I'm worry that such a fetcher would have poor performances

**Reply by gmaze** [reply]
There is also the possibility to simply add the new export button only for data selection that **argopy** can handle !

**Reply by cgourcuf** [reply]
This seems a very good idea from my side too. However, I would recommend to fix pending issues (number of cycles/profiles, etc.) as priorities (see https://gitlab.ifremer.fr/ea-data-selection/ea-data-selection/-/issues)

**Comment by gmaze** [comment]
# 💡 New export button in a profile visualisation page
When user click on a profile from the data selection map, they can open a "Show profile data" link that points to the profile visualisation/dashboard page.
On this page we could also add a new export button:
![Screenshot 2022-03-24 at 15 15 49](https://user-images.githubusercontent.com/1956032/159935951-25fd9e20-2b73-4d87-b3bd-b97601dfe98b.png)



See the new upper right export button proposition.
Clicking this button will “copy” a code snippet to load this profile in python. 

Once clicked, a popup window could be opened to confirm that a code snippet has been “copied” and is ready to be pasted in user code to load this profile
